pulsar-client/src/main/java/org/apache/pulsar/client/api/Authentication.java

```diff
@@ -40,13 +53,46 @@
     AuthenticationDataProvider getAuthData() throws PulsarClientException;
 
     /**
-     * Configure the authentication plugins with the supplied parameters
+     * Configure the authentication plugins with the Map parameter
      *
      * @param authParams
      */
     void configure(Map<String, String> authParams);
 
     /**
+     * Configure the authentication plugins with the String parameter
+     *
+     * @param authParams
+     */
+    default void configure(String authParams) {
+        Map<String, String> authParamsMap = new HashMap<>();
+        Logger LOG = LoggerFactory.getLogger(Authentication.class);
+
+        // Convert JSON to Map
+        if (isNotBlank(authParams)) {
+            try {
+                ObjectMapper jsonMapper = ObjectMapperFactory.create();
+                authParamsMap = jsonMapper.readValue(authParams, new TypeReference<HashMap<String, String>>() {});
+            } catch (JsonParseException e) {
+                // If authParams is not JSON, separate by “:” and “,” for compatibility
+                LOG.warn("Authentication config parameter should be JSON: {}", e.getMessage());
```

Comment:
If we're going to support both comma separated and Json, we shouldn't print a warning here

Reply:
I think that comma separated should be deleted in the future.
What do you think that?

Reply:
In any case the deprecation should be at the interface level, not by having WARN message in log.